File: 이건희/level2/귤 고르기.py
def solution(k, tangerine):
    size = {}
    result = 0
    cnt = 0
    tangerine.sort()
    
    for i in tangerine:
        if i in size:
            size[i] += 1
        else:
            size[i] = 1
            
    size_dic = sorted(size.items(), key=lambda x: x[1], reverse=True)
    
    if size_dic[0][1] >= k:
        return 1
    else:
        for i in size_dic:
            if result < k:
                cnt += 1
                result += i[1]
            else:
                break
                
    return cnt

File: 김찬우/프로그래머스 /Lv.2/귤고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        
        int result = 0;
        int count = 0;
        
        Map<Integer,Integer> map = new HashMap<>();
        
        for(int i=0; i<tangerine.length;i++){
            
            if(map.containsKey(tangerine[i])){
                map.put(tangerine[i],map.get(tangerine[i])+1);
            }else{
                map.put(tangerine[i],1);
            }
            
        }
        
        List<Map.Entry<Integer, Integer>> entryList = new LinkedList<>(map.entrySet());
        entryList.sort(new Comparator<Map.Entry<Integer, Integer>>() {
            
            @Override
            public int compare(Map.Entry<Integer, Integer> o1, Map.Entry<Integer, Integer> o2) {
                return o2.getValue() - o1.getValue();
            }
        });
        
        for(Map.Entry<Integer, Integer> entry : entryList){
            
            result+=entry.getValue();
            
            if(result>=k){
                count++;
                break;
            }
            else{
                count++;
            }
            
        }
            
        return count;
    }
}
